flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/TypeTransformer.java

```diff
@@ -675,4 +686,37 @@ private String getTypeName(Symbol symbol) {
     private List<String> getTypeRefs(Object type, TypeSymbol typeDescriptor) {
         return type instanceof String ? TypeUtils.getTypeRefIds(typeDescriptor, moduleInfo) : List.of();
     }
+
+    private Object handleAsFirstClassNonIntersectionType(IntersectionTypeSymbol intersectionTypeSymbol,
+                                                         TypeData.TypeDataBuilder typeDataBuilder) {
+        SemanticModel semanticModel = module.getCompilation().getSemanticModel();
```

Comment:
`.getCompilation` is not concurrent-safe. If multiple LS requests are sent, there can be intermittent failures.

Reply:
Something similar to this https://github.com/ballerina-platform/ballerina-language-server/blob/fb40fcef6d81b10283e0e6169d3fd40e91ac1c93/model-generator-commons/src/main/java/io/ballerina/modelgenerator/commons/PackageUtil.java#L321

Reply:
Thanks